### DISCUSSION TITLE
Face detection not as good as Deepstack?

### DISCUSSION THREAD

**Discussion by pvi25** [exadel-inc/CompreFace#932] [Q&A]
I have setup my security cam with frigate, doubletake, deepstack and compreface (with MobileNet build). From what I read the face recognition of compreface would be better than that of Deepstack, but for a lot of the images from my camera where a person was detected, Deepstack is able to detect a face, but compreface fails on detecting a face (so can also not recognize it). Is this a known issue and is there a solution for it? (det_prob_threshold is set to 0.4 and opencv_face_required is set to false)

**Comment by pospielov** [comment]
You can lower the `det_prob_threshold` even more, but I do not recommend it, as the false positive increases drastically for bad-quality pictures (I expect that it can't find faces because of the quality)
One more option is to increase the max size of the image that is used for detection. By default, it is 640 pixels, so if the faces are too small, the accuracy will be bad.
We didn't add it to our main functionality yet, but you can look at this pull request:
https://github.com/exadel-inc/CompreFace/pull/800/files
You don't need to rebuild the CompreFace, just change the `docker-compose` and `.env` files.

**Comment by pvi25** [comment]
Lowering gives too many false positives indeed. Added max_detect_size=2048 (various size) to .env but that didnt really make a difference. Attached a screenshot of doubletake to give an impression in which images faces are detected by Deepstack and in which by Compreface (only in 3 of the 13 images where Deepstack detected a face, Compreface also detected a face.


![Test](https://user-images.githubusercontent.com/6019531/198870786-68815668-09e8-45df-be3a-e23e8bf8cd2f.jpg)

**Reply by pospielov** [reply]
There is a chance that mobilenet face detector is simply worse than Deepstack default. In my experience, default CompreFace face detection is better them mobilenet, but default Face recognition is worse than mobilenet. 
I see here only two solutions:
1. Move to a more accurate model, which is arcface
2. Improve the environment - improve light conditions, improve resolution, etc.
